packages/server/src/api/routes/tests/search.spec.ts

```diff
@@ -1950,59 +1950,77 @@ describe.each([
     })
   })
 
-  // This will never work for Lucene.
   !isLucene &&
-    // It also can't work for in-memory searching because the related table name
-    // isn't available.
     !isInMemory &&
     describe("relations", () => {
-      let otherTable: Table
-      let otherRows: Row[]
+      let productCategoryTable: Table, productCatRows: Row[]
 
       beforeAll(async () => {
-        otherTable = await createTable({
-          one: { name: "one", type: FieldType.STRING },
-        })
-        table = await createTable({
-          two: { name: "two", type: FieldType.STRING },
-          other: {
-            type: FieldType.LINK,
-            relationshipType: RelationshipType.ONE_TO_MANY,
-            name: "other",
-            fieldName: "other",
-            tableId: otherTable._id!,
-            constraints: {
-              type: "array",
+        productCategoryTable = await createTable(
+          {
+            name: { name: "name", type: FieldType.STRING },
+          },
+          "productCategory"
+        )
+        table = await createTable(
+          {
+            name: { name: "name", type: FieldType.STRING },
+            productCat: {
+              type: FieldType.LINK,
+              relationshipType: RelationshipType.ONE_TO_MANY,
+              name: "productCat",
+              fieldName: "product",
+              tableId: productCategoryTable._id!,
+              constraints: {
+                type: "array",
+              },
             },
           },
-        })
+          "product"
+        )
 
-        otherRows = await Promise.all([
-          config.api.row.save(otherTable._id!, { one: "foo" }),
-          config.api.row.save(otherTable._id!, { one: "bar" }),
+        productCatRows = await Promise.all([
+          config.api.row.save(productCategoryTable._id!, { name: "foo" }),
+          config.api.row.save(productCategoryTable._id!, { name: "bar" }),
         ])
 
         await Promise.all([
           config.api.row.save(table._id!, {
-            two: "foo",
-            other: [otherRows[0]._id],
+            name: "foo",
+            productCat: [productCatRows[0]._id],
+          }),
+          config.api.row.save(table._id!, {
+            name: "bar",
+            productCat: [productCatRows[1]._id],
           }),
           config.api.row.save(table._id!, {
-            two: "bar",
-            other: [otherRows[1]._id],
+            name: "baz",
+            productCat: [],
           }),
         ])
+      })
 
-        rows = await config.api.row.fetch(table._id!)
+      it("should be able to filter by relationship using column name", async () => {
+        await expectQuery({
+          equal: { ["productCat.name"]: "foo" },
```

Comment:
I think this change makes it possible to run this specific variant of the test against in-memory search. I'm not sure exactly how best to structure the test to do that, but I think it's worth doing.